### PR TITLE
Remove MergedDs

### LINKED ISSUE #973: Remove MergedDs
@egbertbouman There are two download states in `MergedDs`, and I guess the first one is the normal download state and another is for swift. Am I right?

Since we have removed swift, we can also remove this `MergedDs`.

### PR DESCRIPTION
**Pull request by whirm**
A leftover from the swift removal refactor.
Closes #973

### CODE REVIEW

**Comment by LipuFei** [comment]
@whirm a thousand things are using this MergedDs either directly or indirectly. It's a complete maze. I would recommend not to touch it at the moment...

**Comment by whirm** [comment]
@LipuFei they are using lists called dllist but it's not related (yeah, confusing :D)
It seems to be working now, I can download files with no issue. 
Let's see what the tests say...

**Comment by LipuFei** [comment]
@whirm good to know, but be careful. The tests don't test every code that use this MergeDs. Make sure to check the coverage report, especially for those list*.py files.

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@LipuFei yeah :)

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@LipuFei the code coverage looks fine, maybe you want to give it a shake locally before merging?

**Comment by LipuFei** [comment]
@whirm I will test it for a while before merging it.

**Comment by LipuFei** [comment]
Got a backtrace after adding a download:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14665, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/lib/delayedresult.py", line 82, in __call__
    return self.__listener(result, *self.__args, **moreKwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/__init__.py", line 115, in invoke_func
    return func(*args, **kwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/list.py", line 610, in RefreshDelayedData
    self.RefreshData(key, data)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/__init__.py", line 115, in invoke_func
    return func(*args, **kwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/list.py", line 1203, in RefreshData
    original_data.dslist = prevItem.original_data.dslist
AttributeError: 'Torrent' object has no attribute 'dslist'
```
I tries to replace `dslist` with `ds`. It didn't work and showed another backtrace:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14665, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/lib/delayedresult.py", line 82, in __call__
    return self.__listener(result, *self.__args, **moreKwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/__init__.py", line 115, in invoke_func
    return func(*args, **kwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/list.py", line 610, in RefreshDelayedData
    self.RefreshData(key, data)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/__init__.py", line 115, in invoke_func
    return func(*args, **kwargs)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/list.py", line 1203, in RefreshData
    original_data.ds = prevItem.original_data.ds
AttributeError: can't set attribute
```

**Comment by whirm** [comment]
@LipuFei ok, let me fix it, I will be testing the new isolated PR tester in parallel with the old one so we may get a bit spammed by the bot.

**Comment by whirm** [comment]
@LipuFei could you try breaking it again? :D

**Comment by NielsZeilemaker** [comment]
@whirm the ds property in the Torrent class was added to be able to use the mergeds hack.
I would remove it and rename your download_state variable to ds. Then it should be back to normal

**Comment by LipuFei** [comment]
@whirm Got a new one after I added a streaming torrent:
```
Traceback (most recent call last):
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/SearchGridManager.py", line 911, in _do_gui_callback
    callback(dslist, magnetlist)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/EmbeddedPlayer.py", line 182, in OnStatesCallback
    if download_status.get_download() == self.download and self.download.get_mode() == DLMODE_VOD:
AttributeError: 'list' object has no attribute 'get_download'
```

**Comment by LipuFei** [comment]
@whirm Tested at home. Looks good. I will merge it. :)

File: Tribler/Main/Utility/GuiDBTuples.py [hunk 1]
```diff
@@ -92,48 +92,10 @@ def __setstate__(self, statedict):
...
-
-    def get_peerlist(self):
-        return self.dslist[0].get_peerlist() + self.dslist[1].get_peerlist()
-
-
 class Torrent(Helper):
     __slots__ = ('infohash', 'name', 'torrent_file_name', 'length', 'category_id', 'status_id', 'num_seeders',
                  'num_leechers', '_channel', 'channeltorrents_id', 'misc_db', 'torrent_db', 'channelcast_db',
-                 'metadata_db', 'dslist', '_progress', 'relevance_score', 'query_candidates', 'magnetstatus')
+                 'metadata_db', 'ds', '_progress', 'download_status', 'relevance_score', 'query_candidates', 'magnetstatus')
```
Comment (NielsZeilemaker) [inline Tribler/Main/Utility/GuiDBTuples.py]: I would remove ds here

File: Tribler/Main/Utility/GuiDBTuples.py [hunk 2]
```diff
@@ -336,7 +292,7 @@ def __str__(self):
     def __getstate__(self):
         statedict = {}
         for key in Torrent.__slots__:
-            if key not in ['dslist', 'channelcast_db', 'torrent_db']:
+            if key not in ['ds', 'channelcast_db', 'torrent_db']:
```
Comment (NielsZeilemaker) [inline Tribler/Main/Utility/GuiDBTuples.py]: change ds here to download_status

**Comment by NielsZeilemaker** [comment]
I would also rename download_status to download_state as it's not the current status it's a state which was generated in the past

**Comment by synctext** [comment]
@whirm and @LipuFei please take care of Niels his comments.

**Comment by LipuFei** [comment]
@synctext It's done in another PR #1208

**Pull request merged**